db/seeds/api_teachers_with_histories.rb

```diff
@@ -133,3 +139,77 @@ def create_teacher(attrs:)
     end
   end
 end
+
+print_seed_info("Add teachers with changed schedule:")
+contract_period = ContractPeriod.find_by(year: 2022)
+variations = [
+  { change_contract_period: false, change_schedule: true },
+  { change_contract_period: true, change_schedule: false },
+  { change_contract_period: true, change_schedule: true },
+]
+ActiveLeadProvider.where(contract_period:).find_each do |active_lead_provider|
+  %i[ect mentor].each do |trainee_type|
+    variations.each do |variation|
+      teacher = create_teacher(attrs: {})
+      full_name = "#{teacher.trs_first_name} #{teacher.trs_last_name}"
+      print_seed_info("#{full_name} (#{trainee_type.upcase}) - #{variation.keep_if { |_k, v| v }.keys.join(', ')}", indent: 2, colour: (trainee_type == :ect ? :magenta : :yellow))
+
+      started_on = Date.new(contract_period.year, 9, 1) + rand(1..100).days
+      at_school_period = FactoryBot.create(
+        :"#{trainee_type}_at_school_period",
+        teacher:,
+        started_on:,
+        finished_on: nil
+      ).tap { |sp| send(:"describe_#{trainee_type}_at_school_period", sp) }
+
+      school_partnership = random_school_partnership(active_lead_provider:)
+      schedule = random_schedule(contract_period:, trainee_type:)
+      finished_on = at_school_period.started_on + rand(1..100).days
+
+      # Old training period
+      FactoryBot.create(
+        :training_period,
+        :"for_#{trainee_type}",
+        :provider_led,
+        :with_schedule,
```

Comment:
We can drop `with_schedule` on these training period factory calls as we pass it manually